numpy/core/tests/test_arrayprint.py

```diff
@@ -237,12 +243,26 @@ def test_formatter_reset(self):
         assert_equal(repr(x), "array([ 0.,  1.,  2.])")
 
     def test_0d_arrays(self):
+        if sys.version_info[0] >= 3:
+            assert_equal(str(np.array('café', np.unicode_)), 'café')
+            assert_equal(repr(np.array('café', np.unicode_)),
+                         "array('café',\n      dtype='<U4')")
```

Comment:
I always assumed that was just intended to handle the subset which is structured dtypes (which "flexible" is often used to mean exclusively, eg in comments `np.ma`). Certainly for scalars, it seems overkill. Either way, out of scope to change here.